src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -54,6 +59,41 @@ public static string LocalSlotName
 
 #pragma warning disable SA1117 // Parameters should be on same line or separate lines
 
+        private string SanitizeString(string? rawPayload, out string iloggerPayload, bool isReplay = false)
+        {
+            if (isReplay)
+            {
+                iloggerPayload = rawPayload ?? "(replay)";
+                return "(replay)";
+            }
+
+            string payload = rawPayload ?? "(null)";
+            int numCharacters = payload.Length;
+            string sanitizedPayload = $"(Redacted {numCharacters} characters)";
+
+            iloggerPayload = this.shouldTraceRawData ? payload : sanitizedPayload;
+            return sanitizedPayload;
+        }
+
+        private string SanitizeException(Exception? exception, out string iloggerExceptionString, bool isReplay = false)
```

Comment:
The design of this API is a bit confusing. It's not clear what the difference is supposed to be between the output (return value) and the `out` variable. Rather than having a named `out` parameter and an unnamed return value, it would be better from a code understandability perspective to have two named `out` parameters and just return `void`. Having this kind of usage clarity is especially important in this case because making a mistake in interpreting the outputs of this method could cause us to accidentally leak sensitive information.

Reply:
Good point - I have refactored the implementation to use two `out` variables.